controllers/complianceeventsapi/server.go

```diff
@@ -380,9 +426,108 @@ func parseQueryArgs(queryArgs url.Values) (*queryOptions, error) {
 		}
 	}
 
+	authorizedClusterNames, includeAllManagedClusters,
+		err := getAuthorizedClusterNames(ctx, db, parsed, userChangedConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if includeAllManagedClusters {
+		delete(parsed.Filters, "clusters.cluster_id")
+		delete(parsed.Filters, "clusters.name")
+
+		return parsed, nil
+	}
+
+	if len(authorizedClusterNames) == 0 {
+		log.V(1).Info("The service account does not have GET authorization for all")
+
+		return nil, ErrNotAuthorized
+	}
+
+	delete(parsed.Filters, "clusters.cluster_id")
+
+	parsed.Filters["clusters.name"] = authorizedClusterNames
+
 	return parsed, nil
 }
 
+// Return authorized cluster names or, If the url query doesn't have specific clusters, it verifies
+// all managedCluster Authorizations and it returns true if the user has all authorization for manageClusters
+func getAuthorizedClusterNames(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userChangedConfig *rest.Config,
+) ([]string, bool, error) {
+	var authorizedClusterNames []string
+
+	clusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	if okClusterIds {
+		clusterNames := make([]string, 0, len(clusterIDs))
+		// Convert id to name
+		for _, value := range clusterIDs {
+			clusterNames = append(clusterNames, getClusterNameFromID(ctx, db, value))
+		}
+
+		allRules, err := getManagedClusterRules(userChangedConfig, clusterNames)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if slices.Contains(allRules["*"], "get") {
+			authorizedClusterNames = append(authorizedClusterNames, clusterNames...)
+		} else {
+			for _, cn := range clusterNames {
+				if slices.Contains(allRules[cn], "get") || slices.Contains(allRules[cn], "*") {
+					authorizedClusterNames = append(authorizedClusterNames, cn)
+				} else {
+					log.V(1).Info("The service account does not have GET authorization for " + cn)
+				}
+			}
+		}
+	}
+
+	clusterNames, okClusterNames := parsed.Filters["clusters.name"]
+	if okClusterNames {
+		allRules, err := getManagedClusterRules(userChangedConfig, clusterNames)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if slices.Contains(allRules["*"], "get") {
```

Comment:
This is repeating the logic from above. Could you populate a `clusterNames` slice with all the clusters from the `clusters.cluster_id` and `clusters.name` filter and then check `allRules` against it?